src/txn/txn_recover.c

```diff
@@ -592,6 +602,12 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
 	 * First, do a pass through the log to recover the metadata, and
 	 * establish the last checkpoint LSN.  Skip this when opening a hot
 	 * backup: we already have the correct metadata in that case.
+	 *
+	 * If we're running with salvage and we hit an error, we ignore it
+	 * and continue. In salvage we want to recover whatever part of the
+	 * data we can from the last checkpoint up until whatever problem we
+	 * detect in the log file. In salvage we only care about errors from
+	 * scanning the log. Other errors remain errors.
```

Comment:
I think the second to last sentence of this comment doesn't make sense with my read of the code.  Looks like when salvaging we are ignoring errors from scanning the log.